src/main/java/it/gov/pagopa/payhub/ionotification/service/ioservice/IODuplicateServiceHandlerServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package it.gov.pagopa.payhub.ionotification.service.ioservice;
+
+import it.gov.pagopa.payhub.ionotification.connector.IORestConnector;
+import it.gov.pagopa.payhub.ionotification.dto.ServicePaginatedResponseDTO;
+import it.gov.pagopa.payhub.ionotification.dto.ServicesListDTO;
+import it.gov.pagopa.payhub.ionotification.model.IOService;
+import it.gov.pagopa.payhub.model.generated.ServiceRequestDTO;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+@Slf4j
+public class IODuplicateServiceHandlerServiceImpl implements  IODuplicateServiceHandlerService {
+
+    private final IORestConnector connector;
+
+    public IODuplicateServiceHandlerServiceImpl(IORestConnector connector) {
+        this.connector = connector;
+    }
+
+    @Override
+    public String handleDuplicateService(IOService service, ServiceRequestDTO serviceRequestDTO) {
```

Comment:
The scope of this Service is not more the handling of the duplicate service, but just a search operation, let's rename the class and the method
```suggestion
    public String searchIOService(IOService service, ServiceRequestDTO serviceRequestDTO) {
```